File: app/src/main/java/cf/movie/slmovie/application/MyApplication.java
package cf.movie.slmovie.application;

import android.app.Application;

import com.facebook.drawee.backends.pipeline.Fresco;
import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;

import java.util.Arrays;
import java.util.List;

/**
 * Created by 包俊 on 2017/7/19.
 */

public class MyApplication extends Application{

    public static MyApplication mApp;

    @Override
    public void onCreate() {
        mApp = this;
        super.onCreate();
        Fresco.initialize(this);
    }
}

File: app/src/main/java/cf/movie/slmovie/application/MyApplication.kt
package cf.movie.slmovie.application

import android.app.Application
import android.content.pm.PackageManager
import android.util.Log

/**
 * Created by 包俊 on 2017/7/19.
 */

class MyApplication : Application() {
    var instance: MyApplication? = null

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    override fun getPackageName(): String {
        return if (Log.getStackTraceString(Throwable()).contains("com.xunlei.downloadlib")) {
            "com.xunlei.downloadprovider"
        } else super.getPackageName()
    }

    override fun getPackageManager(): PackageManager {
        return if (Log.getStackTraceString(Throwable()).contains("com.xunlei.downloadlib")) {
            DelegateApplicationPackageManager(super.getPackageManager())
        } else super.getPackageManager()
    }
}
